### DISCUSSION TITLE
Prior austenite grain reconstruction by TEM mapping

### DISCUSSION THREAD

**Discussion by viviannh81** [mtex-toolbox/mtex#818] [Ask Anything]
Hello everyone, 

I am working on austenite parent grain reconstruction from mapping performed by TEM (PACOM). The material is tempered martensite and the thin foil was prepared using site-specific FIB-cutting to have at least one previous austenite grain boundary.. I am using mtex 5.1.1 with the reconstruction code by https://github.com/nyyssont/parent_austenite_reconstruction

This is the microstructure observed by TEM (STEM BF) and its IPF
![image](https://user-images.githubusercontent.com/51818024/109157624-8c5fd580-7772-11eb-89ec-7fee438dc984.png)

We saw that depending on the grains identified initially in calcGrains ([grains,ebsd.grainId] = calcGrains(ebsd(string_ferrite),'angle',3*degree);), the reconstruction results are very different. The nature of the grain (austenite gb/packet gb/…etc) and the austenite grain orientations could all be different. (At this scale, the carbides are visible but considered as artifacts for the reconstruction.)

Here is a reconstruction using 3° (black = previous austenite gb, green = packet gb, red = block gb)
![image](https://user-images.githubusercontent.com/51818024/109157150-f3c95580-7771-11eb-856d-9ca8bd270566.png)

And using 0.5° (black = previous austenite gb, green = packet gb, red = block gb)
![image](https://user-images.githubusercontent.com/51818024/109157288-1e1b1300-7772-11eb-9bd3-a551f9f10ef9.png)

We are wondering if this is because there are very few grains identified in a thin foil, so that the reconstruction results strongly depends on the initial grain identification? We would like to ask for help to see if we can obtain proper reconstruction results based on TEM mapping, please. Thank you.

**Comment by nyyssont** [comment]
_We are wondering if this is because there are very few grains identified in a thin foil, so that the reconstruction results strongly depends on the initial grain identification?_

I'd say you are right on the money there. For one thing, it becomes very difficult to get a good representative OR using the automated iterative algorithm when there are only a few orientation domains (martensitic laths and blocks called grains by MTEX) available. This shows in the first row of images, where only a few grain boundaries are identified as block and packet boundaries, and they appear to be misindexations. So it is likely that the algorithm is working with a bad representative OR.

On the second row, on the other hand, the block and packet indexation seems to be much more credible based on visual inspection, which could mean a better representative OR has been found. This is likely thanks to the larger amount of orientation domains available.

So, I would say that the second result is much more likely to represent the prior structure correctly.

If you are interested in improving accuracy, you could produce a larger dataset by EBSD and determining the orientation relationship separately using that and then carrying it to this analysis.

Another thing to consider is that you are likely working at a level where you are close to identifying individual laths. Thus, even low-misorientation angles between laths within blocks become meaningful, as shown by your better results using a 0.5° threshold. How to identify these low-misorientation structures? Difficult question to answer.

Finally, if possible, I would recommend trying out the reconstruction features in MTEX 5.6.0. They are fast and you have a larger toolbox for working around reconstruction-specific problems. After reconstruction, you can handily identify individual laths.

-Tuomo

**Reply by viviannh81** [reply]
Dear Tuomo,

Thank you for the fast respond. I am trying the reconstruction features in MTEX 5.6.0 for a few days, following the tutorial https://mtex-toolbox.github.io/MaParentGrainReconstruction.html

If I understood your proposition correctly as to identify the OR from a large EBSD dataset then use it to reconstruct a smaller ASTAR dataset.
I can identify a job1 for EBSD reconstruction :
job1 = parentGrainReconstructor(ebsd1,grains1);
job1.p2c = orientation.KurdjumovSachs(job1.csParent, job1.csChild)
job1.calcParent2Child %the misfit is optimized

Then identify a job2 for ASTAR reconstruction using the OR identified in job1:
job2 = parentGrainReconstructor(ebsd2,grains2,job1.p2c);
Then proceed with the reconstruction procedure (as indicated in the tutorial) with calculating the misfit between the child pahse : [fit,c2cPairs] = job2.calcGBFit; ...etc
Then the misfit calculation and the reconstruction will all be based on job1.p2c as optimized from the large EBSD dataset?

I am not sure if i understood correctly how am I able to identify individual laths (or blocks). If I understood correctly, in the reconstruction features, maybe they are scales too small for large EBSD datasets?

Thanks

**Reply by nyyssont** [reply]
You describe the procedure I suggested correctly.

On the identification of individual laths: 

Quite often the orientation domains (called grains in MTEX) defined using calcGrains contain several martensitic variants (individual laths) with a very low mutual misorientation angle. Using calcGrains, it is often impossible to separate these variants from each other, no matter how low the threshold angle used for grain identification.

The code below visualizes this using the demo martensite dataset. The resulting image will show the grain boundaries determined by calcGrains, as well as each individual lath ID determined by calcChildVariant. Basically, each color corresponds to a distinct variant.

You should be able to see that within the boundaries of a single grain determined by calcGrains, several different martensitic variants can in fact be found. 

```
mtexdata martensite 
plotx2east

[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',3*degree);
ebsd(grains(grains.grainSize < 3)).rotations = quaternion.nan;
ebsd(grains(grains.grainSize < 3)).phase = 0;
ebsd(grains(grains.grainSize < 3)).grainId = 0;
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',3*degree);

job = parentGrainReconstructor(ebsd,grains) % define job
job.calcParent2Child; % get representative OR

job.calcGBVotes % compute votes from child grain boundaries
job.calcParentFromVote('minVotes',4)

for l = 1:2
job.calcGBVotes('noC2C') % compute votes ONLY from parent-child boundaries;
job.calcParentFromVote('minFit',5*degree) %compute parent orientations from votes
end

parentEBSD = job.calcParentEBSD;
idvar = ~isnan(parentEBSD.fit);
varId = calcChildVariant(parentEBSD(idvar).orientations,ebsd(idvar).orientations,job.p2c);

plot(parentEBSD(idvar),varId)
hold on
plot(grains.boundary,'linewidth',2)
```
